docs/specs/openapi.yaml

```
@@ -136,18 +339,227 @@ paths:
+          description: Action to perform.
+          required: true
+          schema:
+            type: string
+            enum: ["list", "read"]
+        - name: path
+          in: query
+          description: |
+            For "read": Absolute file path to read. To read multiple files, specify this parameter multiple times.
+            
+            For "list": Absolute path to the directory to list.
+          required: true
+          schema:
+            type: string
+          style: form # For handling comma-separated values in "read".
+          explode: true # For handling comma-separated values in "read".
+        - name: pattern
+          in: query
+          description: Glob pattern to filter files/directories for the "list" action.
+          schema:
+            type: string
+        - name: itself
+          in: query
+          description: Whether to list the directory itself, rather than its contents for the "list" action.
+          schema:
+            type: string
+            enum: ["false", "true"]
+      responses:
+        "200":
+          description: |
+            For "read":  Multipart form data response with file contents and metadata.
+
+            For "list": JSON array of file information.
+          content:
+            application/json: # list
+              schema:
+                $ref: "#/components/schemas/ListFilesResponse"
+              example:
+                {
+                  "type": "sync",
+                  "status-code": 200,
+                  "status": "OK",
+                  "result": [
+                    {
+                      "path": "/home/ubuntu/PEBBLE_HOME/layers/001-simple-layer.yaml",
+                      "name": "001-simple-layer.yaml",
+                      "type": "file",
+                      "size": 122,
+                      "permissions": "664",
+                      "last-modified": "2024-12-27T11:13:31+08:00",
+                      "user-id": 1000,
+                      "user": "ubuntu",
+                      "group-id": 1000,
+                      "group": "ubuntu"
+                    }
+                  ]
+                }
+            multipart/form-data: # read
+              schema:
+                $ref: "#/components/schemas/ReadFilesResponse"
+              example:
+                files:
+                  "foo some file content"
+                response:
+                  {
+                    "type": "sync",
+                    "status-code": 200,
+                    "status": "OK",
+                    "result": [
+                      {
+                        "path": "/home/ubuntu/bar"
+                      }
+                    ]
+                  }
+    post:
+      summary: Create, write, remove files/directories.
+      tags:
+        - files
+      description: |
+        Write content to a path on the remote system;
+        
+        creates a directory or directory tree;
+
+        delete a file or directory.
```

Suggest writing this as bullets:
```suggestion
        This endpoint can:
        - Write content to a path on the remote system        
        - Create a directory or directory tree
        - Delete a file or directory
```